app/services/data_sync/elasticsearch/user.rb

```diff
@@ -0,0 +1,51 @@
+module DataSync
+  module Elasticsearch
+    class User
+      RELATED_DOCS = %i[
+        articles
+        reactions
+        podcast_episodes
+        chat_channel_memberships
+        comments
+      ].freeze
+
+      SHARED_USER_FIELDS = %i[
+        username
+        name
+        pro
+        profile_image_url
+      ].freeze
+
+      attr_accessor :user, :updated_fields
+
+      delegate :articles, :chat_channel_memberships, :comments, to: :@user
+
+      def initialize(user, updated_fields)
+        @user = user
+        @updated_fields = updated_fields.deep_symbolize_keys
+      end
+
+      def sync_documents
+        return unless sync_needed?
+
+        RELATED_DOCS.each do |relation_name|
```

Comment:
I could see this being extracted to a superclass or module if more DataSync classes are added!